### ISSUE TITLE
Broken events if jQuery object was stored by referense, removed and appended again

### ISSUE THREAD

**Issue by hastom** [jquery/jquery#3776]
Seems to me like unexpected behavior.
Steps to reproduce:

1. Create jQuery object.
2. Attach any event to it
3. Store jQuery object inside any object
4. Append it to dom from object
5. Remove it from dom
6. Append again

Expected:
Event still working
Got:
Event no longer triggers

Example: https://codepen.io/hastom/pen/dzBrVd

**Comment by gibson042** [comment]
http://api.jquery.com/remove/ :
> In addition to the elements themselves, all bound events and jQuery data associated with the elements are removed. To remove the elements without removing data and events, use `.detach()` instead.
